core/src/commitment_service.rs

```diff
@@ -127,6 +139,7 @@ impl AggregateCommitmentService {
         block_commitment_cache: &RwLock<BlockCommitmentCache>,
         aggregation_data: CommitmentAggregationData,
         ancestors: Vec<u64>,
+        _bank_forks: &Arc<RwLock<BankForks>>,
```

Comment:
This is the key change. By passing in BankForks here, we can use it in ReplayStake to get all the Banks. By using Banks in CommitmentCache instead of slots, the RPC service should never need to grab a BankForks lock.